plugins/destination/bigquery/client/spec.go

```diff
@@ -34,6 +34,9 @@ type Spec struct {
 	DatasetLocation       string                 `json:"dataset_location"`
 	TimePartitioning      TimePartitioningOption `json:"time_partitioning"`
 	ServiceAccountKeyJSON string                 `json:"service_account_key_json"`
+
+	// We override this in tests to ensure schema is ready
+	MinTriesSleepTimeSeconds int `json:"min_tries_sleep_time_seconds"`
```

Comment:
If that's a test-only field, don't export it & fill it manually

Reply:
I tried, couldn't find another way to pass configuration to plugins except the spec

Reply:
It's all in the same package, so think it can still be made private, so it doesn't become accessible to users in the yaml?

Reply:
Sure I can do that (a global var), wasn't sure if it would be cleaner, but will give it a shot

Reply:
It doesn't have to be a global var, it can be a private value in the spec, right?

Reply:
I tried it but it doesn't propagate to the plugin as we do `UnmarshalSpec` during init so any non exported fields are missing after that

Reply:
See relevant code https://github.com/cloudquery/cloudquery/blob/da5d2f8a4a0aae5f16905224e4901f64c733ca4d/plugins/destination/bigquery/client/client.go#L34.

If it's not exported in the struct via a JSON tag, we won't have that value in the plugin spec

Reply:
Ah okay, I see, thanks! Maybe an unexported global var is the way to go then: even though that's not great either, at least it doesn't impact user-facing config then

Reply:
Moved it to a global var but happy to do any other working solution.
Maybe another way to solve the delay is using an emulator (e.g. https://github.com/goccy/bigquery-emulator) instead of the real service for tests. Maybe then the schema update is instant

Reply:
Ah yeah, I thought about that too but then saw that there was no official bigquery emulator. I didn't see this unofficial one, that looks like a great option!

Reply:
Opened https://github.com/cloudquery/cloudquery/issues/8258 with some more context

Reply:
@candiduslynx do you have any concerns with this PR? You review is blocking it

Reply:
approved